applets/clock/clock.c

```diff
@@ -146,6 +149,9 @@ struct _ClockData {
         ClockFormat  format;
         char        *custom_format;
         gboolean     showseconds;
+        gboolean     showlongdate;
+	gboolean     showshortdate;
+        gboolean     showdefaultdate;
```

Comment:
You really only need a boolean variable, for instance showlongdate and set its default value.